ti-admob/index.js

```diff
@@ -0,0 +1,105 @@
+const mobads = require('com.google.android.gms.ads.MobileAds');
+const Activity = require('android.app.Activity');
+const InterstitialAd = require('com.google.android.gms.ads.InterstitialAd');
+const AdRequest = require('com.google.android.gms.ads.AdRequest');
+const AdListener = require('com.google.android.gms.ads.AdListener');
+
+let hasInitApp = false;
+
+const ADTYPES = {
+    AD_TYPE_INTERSTITIAL: 0
+};
+
+class AdView {
+    constructor(options) {
+        this.adView = null;
+        this.options = options;
+        this.activity = new Activity(Ti.Android.currentActivity);
+        if (!hasInitApp && !this.options.appId) {
+            // setting default/test appId by google
+            this.options.appId = 'ca-app-pub-3940256099942544~3347511713'; 
+            Ti.API.warn('Default/Test AppId is used that is provided by Google');
+        }
+
+        if (!hasInitApp) {
+            this.appId = this.options.appId;
+            hasInitApp = true;
+        }
+        
+        if (!this.options.adId) {
+            this.options.adId = 'ca-app-pub-3940256099942544/1033173712'; 
+            Ti.API.warn('Default/Test AdId is used that is provided by Google');
+        }
+        
+        this.createAd();
+    }
+    
+    createAd() {
+        if (this.options.adType === ADTYPES.AD_TYPE_INTERSTITIAL && Ti.Android) {
+            this.createInterstitialAdAndroid();
+        }
+        
+        if (this.adView !== null) {
+            // first time, load already
+            this.load();
+        }
+    }
+    
+    load() {
+        this.adView.loadAd(new AdRequest.Builder().build());
+    }
+    
+    createInterstitialAdAndroid() {
+        this.adView = new InterstitialAd(this.activity);
+        this.adView.setAdUnitId(this.options.adId); // replace ad ID with your own
+        this.adView.setAdListener(this.AdListener);        
+    }
+    
+    get AdListener() {
+        let self = this;
+        let ADL = com.google.android.gms.ads.AdListener.extend({
+            onAdLoaded: () => {
+                // add loaded, now ready to be shown
+                if (self.options.onAdLoaded) self.options.onAdLoaded();
+            },
+            onAdFailedToLoad: (e) => {
```

Comment:
Unused event. Should likely be unwrapped and passed to the client to handle the error.